### ISSUE TITLE
kill-session on Mac M1 does not quit and maxes CPU usage

### ISSUE THREAD

**Issue by WizardOfMenlo** [zellij-org/zellij#1440]
Please attach the files that were created in `/tmp/zellij-501/zellij-log/` to the extent you are comfortable with.
```
INFO   |zellij_client            | 2022-05-30 14:31:48.895 [main      ] [/Users/wiz/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-client-0.29.1/src/lib.rs:123]: Starting Zellij client! 
INFO   |zellij_server            | 2022-05-30 14:31:48.900 [main      ] [/Users/wiz/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.29.1/src/lib.rs:189]: Starting Zellij server! 
INFO   |zellij_server::wasm_vm   | 2022-05-30 14:31:48.957 [wasm      ] [/Users/wiz/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.29.1/src/wasm_vm.rs:87]: Wasm main thread starts 
INFO   |zellij_client            | 2022-05-30 14:32:00.458 [main      ] [/Users/wiz/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-client-0.29.1/src/lib.rs:123]: Starting Zellij client! 
INFO   |zellij_server            | 2022-05-30 14:32:00.463 [main      ] [/Users/wiz/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.29.1/src/lib.rs:189]: Starting Zellij server! 
INFO   |zellij_server::wasm_vm   | 2022-05-30 14:32:00.519 [wasm      ] [/Users/wiz/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.29.1/src/wasm_vm.rs:87]: Wasm main thread starts 
INFO   |zellij_client            | 2022-05-30 14:32:19.753 [main      ] [/Users/wiz/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-client-0.29.1/src/lib.rs:366]: Bye from Zellij! 
INFO   |zellij_server::wasm_vm   | 2022-05-30 14:32:19.753 [wasm      ] [/Users/wiz/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.29.1/src/wasm_vm.rs:213]: wasm main thread exits 
ERROR  |zellij_server::route     | 2022-05-30 14:32:19.754 [server_router] [/Users/wiz/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.29.1/src/route.rs:510]: Received empty message from client 
ERROR  |zellij_server::route     | 2022-05-30 14:32:19.754 [server_router] [/Users/wiz/.cargo/registry/src/github.com-1ecc6299db9ec823/zellij-server-0.29.1/src/route.rs:510]: Received empty message from client
... Approximately 76 MB of the same error message
```

**Basic information**

`zellij --version`: `zellij 0.29.1`
`tput lines`: `30`
`tput cols`: `92`
`uname -av`: `Darwin wiz-MacBook-Pro.local 21.3.0 Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T6000 arm6`

List of programs you interact with as, `PROGRAM --version`: output cropped meaningful, for example:
`nvim --version`: `NVIM v0.7.0`
`alacritty --version`: `alacritty 0.10.1 (2844606)`
`nu --version`: `0.61.0`

**Further information**
To reproduce, start a new Zellij session with `zellij`. From a different terminal run `zellij list-sessions` and `zellij kill-session` with the resulting session identifier. The session will be killed but zellij then does not fully exits and maxes out your CPU. If you have many open and then do `zellij kill-all-sessions` your Mac will also become very warm. It seems that you then have to Force Quit to reclaim back your resources.

**Comment by tlinford** [comment]
I can't reproduce this anymore on v0.30.0, probably was fixed by #1454.
@WizardOfMenlo could you give it a try and confirm the issue is resolved?

**Comment by WizardOfMenlo** [comment]
Hi, can confirm that 0.30.0 seems to have fixed this.
